### PR TITLE
Replace `NoReturn` with `None`

### PR DESCRIPTION
**Pull request by LiHRaM**
Hey, looks like GitHub has some strange behavior around force pushing, so I'm attempting again. 😄 

This PR is a second attempt at #91, but replaces `NoReturn` with `None`.

Regarding the documentation, I was unable to run `make github` with any version of sphinx / rtd theme that didn't add files. Not sure what to do about that.

- [x] Checked that tests passed using `python -m unittest` (python 3.9)
- [x] Replaced `NoReturn` with `None`
- [x] Add entry to CHANGELOG.txt with attribution

### CODE REVIEW

File: docsrc/new_bandit.rst [hunk 1]
```diff
@@ -113,7 +113,7 @@ Here is, at a high-level, what you need to implement in your bandit policy:
             # These fields are declared here and initialized to zero.
             self.my_value_to_arm = dict.fromkeys(self.arms, 0)
 
-        def fit(self, decisions: np.ndarray, rewards: np.ndarray, contexts: np.ndarray = None) -> NoReturn:
```
Comment (skadio) [inline docsrc/new_bandit.rst:116]: thank you for also updating the docs!

**Review by skadio — APPROVED** [review]

File: docsrc/new_bandit.rst (hunk 1 above)
Comment (LiHRaM) [reply docsrc/new_bandit.rst:116]: No problem :)

**Comment by LiHRaM** [comment]
Hey @skadio, as I'm unable to merge I just wanted to check in on the status of this PR. Is there anything else that needs to be done from my side?

I had a look through the differences in documentation and I can see some things have changed. I'll add the examples below this line.

---

**Example 1**: You can see the code formatting is slightly different.

image


**Example 2**: The info boxes are rendered differently

image


**Example 3**: Lists are rendered differently

image

**Comment by LiHRaM** [comment]
Hey @skadio, since I can't merge this request myself, I just wanted to check in. Is there anything else that needs to be done from my side?

**Comment by skadio** [comment]
@LiHRaM This PR is good to go, we just wanted to check a few things outside the library, in case the return change had an impact.  We will merge shortly

**Pull request merged**